jep/0000/README.adoc

```diff
@@ -0,0 +1,246 @@
+| Status
+| Not Submitted :information_source:
+
+| Type
+| Standards
+
+| Created
+| 2018-06-22
+
+| BDFL-Delegate
+| :bulb: Link to github user page :bulb:
+
+| JIRA
+| link:https://issues.jenkins-ci.org/browse/JENKINS-31096[JENKINS-31096]
+//
+//
+// Uncomment if discussion will occur in forum other than jenkinsci-dev@ mailing list.
+//| Discussions-To
+//| :bulb: Link to where discussion and final status announcement will occur :bulb:
+//
+//
+// Uncomment if this JEP depends on one or more other JEPs.
+//| Requires
+//| :bulb: JEP-NUMBER, JEP-NUMBER... :bulb:
+//
+//
+// Uncomment and fill if this JEP is rendered obsolete by a later JEP
+//| Superseded-By
+//| :bulb: JEP-NUMBER :bulb:
+//
+//
+// Uncomment when this JEP status is set to Accepted, Rejected or Withdrawn.
+//| Resolution
+//| :bulb: Link to relevant post in the jenkinsci-dev@ mailing list archives :bulb:
+
+|===
+
+== Abstract
+
+All Jenkins Pipeline build logs are stored internally in UTF-8 encoding.
+(The log can be seen by users from the *Console* link,
+but also from Blue Ocean, *Pipeline Steps*, and REST and CLI.)
+
+Text produced by Jenkins (Java) code is intrinsically Unicode.
+Output from external processes which was not already in UTF-8 is converted to UTF-8 as it is produced.
+
+The scope does not include other project types (freestyle),
+or other types of logging (system, branch indexing).
+
+== Specification
+
+Newly created Pipeline build logs use UTF-8 encoding for all text.
+This is hard-coded in the `charset` field in `build.xml`.
+The `TaskListener` associated with the build as a whole (`FlowExecutionOwner.getListener`),
+and with individual steps (`StepContext.get(TaskListener.class)`),
+will thus be using UTF-8 encoding.
+
+When output is produced by `DurableTaskStep` (`sh`, `bat`, `powershell`),
+it is transcoded to UTF-8 where necessary by `FileMonitoringController`.
+The API methods `charset` and `defaultCharset` in the `DurableTask` interface support this.
+The source encoding defaults to the system default encoding of the node (`Charset.defaultCharset`),
+but may be explicitly specified with a `encoding` parameter.
+
+== Motivation
+
+Jenkins freestyle build records retain a character encoding field `charset`,
+which may vary from build to build.
+Since a freestyle build always allocates exactly one node,
+the encoding is just the system default encoding of that node,
+and the build log is understood to use that encoding.
+Output from build steps (external processes)
+is copied directly to the build log as a bytestream.
+Content rendered over HTTP is always sent in UTF-8 encoding.
+
+When Pipeline was introduced,
+this system could not applied as is
+since a Pipeline build could use many nodes, or none.
+The build log was just left in the _master_’s system default encoding;
+if that encoding happens to be identical to that used by an agent,
+then process output will be encoded directly,
+otherwise you get mojibake.
```

Comment:
Probably worth using a different term than "mojibake" because not all JEPs will be read by people familiar with this class of encoding issue.  I suggest 'garbled text for invalid characters ("mojibake")' to clarify.

Reply:
That is quite a mouthful to repeat. I will try to explain and hyperlink on first usage.